src/args/make-script/template.sh

```diff
@@ -24,6 +24,24 @@ function prompt_user_for_confirmation {
     fi
 }
 
+function prompt_user_for_input {
+  local answer
+  local result="$1"
+  local question="$2"
+
+  warn Please enter your input: \
+    info "$question" \
+    && if running_in_ci_cd_provider; then
+      critical Running on CI/CD provider, cannot provid user input
+    else
+      read -ep '>>> ' -r answer \
+        && eval $result="'$answer'" \
```

Comment:
it seems this `eval` is not needed because $result var is not used elsewhere

Reply:
` local result="$1"`

the trick is: `prompt_user_for_input myresultvar "my question" && echo "$myresult"`

Maybe that can be resolved in a "safer" way while maintaining the ability to echo stuff to stdout?

Reply:
I get the point, what about:


```bash
function prompt_user_for_input {
  local answer
  local default="${1}"

  warn Please type an answer to continue: \
    && if running_in_ci_cd_provider; then
      warn Running on CI/CD provider, assuming you would have answered: "${default}" \
        && echo "${default}"
    else
      read -ep '>>> ' -r answer \
        && echo "${answer}"
    fi
}
```

```bash
info Please enter your age
result=$(prompt_user_for_input "25")
info you are ${result} years old
```

the user is able to modify execution and the CI is able to work with the default value

Reply:
afaik, this would also capture `warn` outputs...

Reply:
The trade off is acceptable:

```
declare -i var
prompt_user_for_input var "question?"
```

... would have errored with a type mismatch telling something abount it's not an integer, if no int was supplied.

---

Now it only fails with a cryptic:

```
[INFO] Token mint quantity
++ prompt_user_for_input
++ local answer
++ local default=
++ warn Please enter your input:
++ echo_stderr '[WARNING]' Please enter your input:
++ echo '[WARNING]' Please enter your input:
[WARNING] Please enter your input:
++ running_in_ci_cd_provider
++ export CI
++ test -n ''
++ read -ep '>>> ' -r answer
>>> sht
++ test -z sht
++ echo sht
+ token_mint_qty=sht
/tmp/nix-shell.VaSzy8/tmplwxc2q_b/result-mintNativeToken-10prepareTransaction/bin/mint-native-tokens-prepare-transaction: line 120: sht: unbound variable
```

Reply:
```
declare -i var
prompt_user_for_input var "question?"

Reply:
this is a good interface IMHO

```
declare -i var
var=$(prompt_user_for_input "question?")
```

Reply:
bash is not well suited for integer aritmetic jeje

I normally handle integers as strings, and operate them as integers with `$(( ... ))` as needed